operations/pyroscope/helm/pyroscope/templates/ingress.yaml

```diff
@@ -11,6 +11,13 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  {{- with .Values.ingress.tls }}
+  tls:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```

Comment:
I'm not very sure we want the capability version check as we unconditionally set the API version to `networking.k8s.io/v1` (1.19)